src/Symfony/Component/Mailer/Transport/Smtp/Stream/SocketStream.php

```diff
@@ -40,7 +40,7 @@ public function setTimeout(float $timeout): self
 
     public function getTimeout(): float
     {
-        return $this->timeout;
+        return $this->timeout ?? $this->timeout = (float) ini_get('default_socket_timeout');
```

Comment:
is freezing the default value required? if not, maybe remove it?

Reply:
we retrieve the timeout twice in `initialize()` ( even if timeout is kept in a local variables, the method can be called more than once, e.g SmtpTransport can restart the socket ), so i think its best to do this and save calling `ini_get` multiple times.

Reply:
There is always a drawback - here, it's giving up the possibility to adopt changes to the ini setting. We need to chose one.